airflow/www/api/experimental/endpoints.py

```diff
@@ -269,6 +270,40 @@ def dag_run_status(dag_id, execution_date):
     return jsonify(info)
 
 
+@csrf.exempt
+@api_experimental.route('/dags/<string:dag_id>/cancel_dag_run', methods=['POST'])
+@requires_authentication
+def cancel_dag_run(dag_id):
+    """
+    Cancel a specific dagrun run_id for a given DAG
+    """
+    data = request.get_json(force=True)
+
+    run_id = None
+    if 'run_id' in data:
```

Comment:
Can we just make this `if run_id not in data raise error` Then just below `_cancel_dag_run(dag_id, data['run_id'])`?

Reply:
Yes will do.